ash/src/device.rs

```diff
@@ -65,6 +69,274 @@ impl Device {
+    ) {
+        (self.device_fn_1_4.cmd_bind_index_buffer2)(
+            command_buffer,
+            buffer,
+            offset,
+            size,
+            index_type,
+        )
+    }
+
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkGetRenderingAreaGranularity.html>
+    #[inline]
+    #[doc(alias = "vkGetRenderingAreaGranularity")]
+    pub unsafe fn get_rendering_area_granularity(
+        &self,
+        rendering_area_info: &vk::RenderingAreaInfo<'_>,
+    ) -> vk::Extent2D {
+        let mut granularity = mem::MaybeUninit::uninit();
+        (self.device_fn_1_4.get_rendering_area_granularity)(
+            self.handle,
+            rendering_area_info,
+            granularity.as_mut_ptr(),
+        );
+        granularity.assume_init()
+    }
+
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkGetDeviceImageSubresourceLayout.html>
+    #[inline]
+    #[doc(alias = "vkGetDeviceImageSubresourceLayout")]
+    pub unsafe fn get_device_image_subresource_layout(
+        &self,
+        info: &vk::DeviceImageSubresourceInfo<'_>,
+        layout: &mut vk::SubresourceLayout2<'_>,
+    ) {
+        (self.device_fn_1_4.get_device_image_subresource_layout)(self.handle, info, layout)
+    }
+
+    // XXX: Replicate comments from VK_EXT_host_image_copy, VK_EXT_image_compression_control
+    // and VK_KHR_maintenance5 about alternative ways to fetch and use a suffixed vesion of this
+    // function?  And retroactively update those docs?
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkGetImageSubresourceLayout2.html>
+    #[inline]
+    #[doc(alias = "vkGetImageSubresourceLayout2")]
+    pub unsafe fn get_image_subresource_layout2(
+        &self,
+        image: vk::Image,
+        subresource: &vk::ImageSubresource2<'_>,
+        layout: &mut vk::SubresourceLayout2<'_>,
+    ) {
+        (self.device_fn_1_4.get_image_subresource_layout2)(self.handle, image, subresource, layout)
+    }
+
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkCmdPushDescriptorSet.html>
+    #[inline]
+    #[doc(alias = "vkCmdPushDescriptorSet")]
+    pub unsafe fn cmd_push_descriptor_set(
+        &self,
+        command_buffer: vk::CommandBuffer,
+        pipeline_bind_point: vk::PipelineBindPoint,
+        layout: vk::PipelineLayout,
+        set: u32,
+        descriptor_writes: &[vk::WriteDescriptorSet<'_>],
+    ) {
+        (self.device_fn_1_4.cmd_push_descriptor_set)(
+            command_buffer,
+            pipeline_bind_point,
+            layout,
+            set,
+            descriptor_writes.len() as u32,
+            descriptor_writes.as_ptr(),
+        )
+    }
+
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkCmdPushDescriptorSetWithTemplate.html>
+    #[inline]
+    #[doc(alias = "vkCmdPushDescriptorSetWithTemplate")]
+    pub unsafe fn cmd_push_descriptor_set_with_template(
```

Comment:
I bet there's a nice type-safe way to wrap this family of functions, though it should be built by someone who actually needs that. Not a blocker.

Reply:
Perhaps yeah. At least we already had this in an extension.